### ISSUE TITLE
Walking sounds when sneaking

### ISSUE THREAD

**Issue by Fixer-007** [minetest/minetest#5651]
Currently if you sneak-walk you can hear walking sounds, but some games may want player to be silent during such slow walks.

**Comment by nerzhul** [comment]
Fix me it sounds like a bug more than a feature, sneaking should reduce sound produced by walk

**Comment by Calinou** [comment]
If this is done, then sneaking status needs to be sent over the network as well, so that you can't hear the sneaking sounds of other players.

**Comment by paramat** [comment]
More of a request than a bug.

**Comment by jastevenson303** [comment]
I personally believe this is an important feature to implement, for what that's worth.  Sorry for the duplicate issue.

**Comment by TekhnaeRaav** [comment]
I think this one is obvious to put in, as long as there is not a ton effort to put into it.

**Comment by MarkuBu** [comment]
The problem is: the engine makes the sound, but speed change is made in players_api. The engine needs to know if the player is sneaking.

**Comment by Calinou** [comment]
> The engine needs to know if the player is sneaking.

It already does, both client-side (to prevent falling from node edges) and server-side (to display the slowed down sneak-walking animation).

**Comment by paramat** [comment]
The important part is not hearing other player's footsteps, that makes this a little more complex.

**Comment by jastevenson303** [comment]
Actually, the important part is not hearing /your own/ footsteps!  The
multiplayer part doesn't matter if I'm not playing multiplayer.

**Comment by Wuzzy2** [comment]
This is an interesting suggestion. Would allow you to hide. Also would give another use to sneaking. Very interesting for PvP.

I'm unsure if this feature should be made optional on the server side. E.g. that mods could force the server to play footstep sounds anyway (in case the server doesn't want to make hiding easy).

IMO this is not very important.

**Comment by rubenwardy** [comment]
This is a nice feature to give feedback to the sneak action (along with sinking the camera, if that's every implemented). I would say it's not very high priority at all, however

**Comment by paramat** [comment]
Muting other player's footsteps is as important as muting your own, otherwise it would be sloppy. So this would mean sending more per-player information to clients.
This is not useful as feedback because you need to know you are sneaking before you sneak, a footstep happening in 1s will be too late if you fall off an edge.

**Comment by Wuzzy2** [comment]
I agree. Either do it right the first timeor don't do it. Not implementing it would still be better than implementing it half-finished and buggy.

**Comment by rubenwardy** [comment]
> This is not useful as feedback because you need to know you are sneaking before you sneak, a footstep happening in 1s will be too late if you fall off an edge.

Ah, that's a good point

**Comment by paramat** [comment]
Yes good feature.

**Comment by jastevenson303** [comment]
Ideally I'd be able to sneak up on another player without them hearing me.  It's currently possible to hide nametag, even the actual player, but not this.  Why should it be considered such "low priority"?

**Comment by Amaz1** [comment]
Can this be solved in lua? Inserting this line [here](https://github.com/minetest/minetest_game/blob/master/mods/player_api/api.lua#L125) seems to work (I've not tested in multiplayer though):
```lua
player:set_properties({makes_footstep_sound = not controls.sneak})
```

Or would this be too intensive/laggy?

**Comment by jastevenson303** [comment]
Wow, so it may have been possible all along!  Will test it out, thanks Amaz.

**Comment by benrob0329** [comment]
That call would likely need to be called every so often (for example, as a globalstep **with time checks**) for it to be updated.

**Comment by jastevenson303** [comment]
Okay, I'll try it and see.

**Comment by jastevenson303** [comment]
Alright, thanks @Amaz1!  It seems to work alright.
```lua
sneak = {}

minetest.register_on_joinplayer(function(player)
	local name = player:get_player_name()
	sneak[name] = false
end)

minetest.register_on_leaveplayer(function(player)
	local name = player:get_player_name()
	sneak[name] = nil
end)

minetest.register_globalstep(function(dtime)
	for _, player in ipairs(minetest.get_connected_players()) do
		local control = player:get_player_control()
		local name = player:get_player_name()
		if control.sneak ~= sneak[name] then
			local c = control.sneak
			if c then
				player:set_properties{makes_footstep_sound = false}
			else
				player:set_properties{makes_footstep_sound = true}
			end
			sneak[name] = c
		end
	end
end)
```
https://github.com/jastevenson303/sneak

**Comment by paramat** [comment]
Good, you're minimising how often 'set properties' is done, that's important.
You could make the check much less often then 10 times a second too.

**Comment by Zughy** [comment]
Bump, sounds make sneaking pretty useless
